sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -500,6 +504,48 @@ def m_text_document__completion(self, **kwargs):
     def m_workspace__execute_command(self, command=None, arguments=()) -> Any:
         return command_dispatcher.dispatch(self, command, arguments)
 
+    @overrides(PythonLanguageServer.m_text_document__did_save)
+    def m_text_document__did_save(self, textDocument=None, **_kwargs):
+        PythonLanguageServer.m_text_document__did_save(self, textDocument, **_kwargs)
+
+        doc_uri = textDocument["uri"]
+
+        fs_path = uris.to_fs_path(doc_uri)
+
+        # Handle Sema4.ai folder files changes with warning message
+        if "Sema4.ai" in fs_path:
```

Comment:
The one detail is that this should also only be done if it's an action package inside an agent package with the `Sema4.ai` being found between those (i.e.: if the user puts everything they do related to Sema4.ai in a root folder, we shouldn't trigger this condition due to that).